### PR TITLE
[Merged by Bors] - feat: Support podOverrides

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/346

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/druid-operator-it-custom/108/

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,10 +10,10 @@ All notable changes to this project will be documented in this file.
 - Missing CRD defaults for `status.conditions` field ([#439]).
 - Support for Druid `26.0.0` ([#442]).
 - Set explicit resources on all containers ([#444]).
+- Support podOverrides ([#XXX]).
```
Comment (maltesander) [inline CHANGELOG.md:13]: PR number and entry missing

File: rust/crd/src/lib.rs [hunk 2]
```diff
@@ -818,6 +821,55 @@ impl DruidCluster {
...
+                .role_groups
+                .get(role_group)
+                .map(|rg| &rg.config.pod_overrides),
+            DruidRole::Router => self
+                .spec
+                .routers
+                .role_groups
+                .get(role_group)
+                .map(|rg| &rg.config.pod_overrides),
+        }
```
Comment (maltesander) [inline rust/crd/src/lib.rs:871]: Could just extract the role_groups in the match and do the get() outside once?

**Review by maltesander — APPROVED** [review]
LGTM, just nitpicking!

File: rust/crd/src/lib.rs (hunk 2 above)
Comment (sbernauer) [reply rust/crd/src/lib.rs:871]: Nope :/
```
error[E0308]: `match` arms have incompatible types
   --> rust/crd/src/lib.rs:842:39
    |
840 | /         match role {
841 | |             DruidRole::Broker => self.spec.brokers.role_groups,
    | |                                  ----------------------------- this is found to be of type `HashMap<std::string::String, RoleGroup<BrokerConfigFragment>>`
842 | |             DruidRole::Coordinator => self.spec.coordinators.role_groups,
    | |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `BrokerConfigFragment`, found struct `CoordinatorConfigFragment`
843 | |             DruidRole::Historical => self.spec.historicals.role_groups,
844 | |             DruidRole::MiddleManager => self.spec.middle_managers.role_groups,
845 | |             DruidRole::Router => self.spec.routers.role_groups,
846 | |         }
    | |_________- `match` arms have incompatible types
    |
    = note: expected struct `HashMap<_, RoleGroup<BrokerConfigFragment>>`
               found struct `HashMap<_, RoleGroup<CoordinatorConfigFragment>>
````

Comment (maltesander) [reply rust/crd/src/lib.rs:871]: ok resolved! :D

**Comment by sbernauer** [comment]
@maltesander please have a look at the new test

**Comment by sbernauer** [comment]
Let's use Azure https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/druid-operator-it-custom/109/

**Comment by sbernauer** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/druid-operator-it-custom/110/

**Comment by maltesander** [comment]
Looks good!

**Comment by sbernauer** [comment]
bors r+
Resources test failed, fixed it and passed locally

**Pull request closed without merging**